src/Microsoft.IdentityModel.Tokens.Saml/Saml2/Saml2Conditions.cs

```diff
@@ -82,12 +82,12 @@ public DateTime? NotOnOrAfter
             {
                 value = DateTimeUtil.ToUniversalTime(value);
 
-                // NotBefore must be earlier than NotOnOrAfter
-                if (value != null && NotBefore.HasValue)
-                {
-                    if (value.Value <= NotBefore.Value)
-                        throw LogExceptionMessage(new ArgumentException(FormatInvariant(LogMessages.IDX13514, MarkAsNonPII(value), MarkAsNonPII(NotBefore))));
-                }
+                //This should not be checked here, instead fail during validation of the token. Will remove this code once we release new validation model bug #2905
```

Comment:
nit: I usually find more clear the single line `//` comments as they are easier to know where they begin and end when looking at git history or PRs.

Reply:
In bug #2905 can you make sure this is linked there?